dao-go/assignments_test.go

```diff
@@ -2,14 +2,337 @@ package dao_test
+	assetStr := asset.Impl.(*eos.Asset).String()
+	stringItms := strings.Split(assetStr, " ")
+	parsed, err := strconv.ParseFloat(stringItms[0], 32)
+
+	return int64(parsed), stringItms[1], err
+}
+
+func ValidateLastReceipt(targetHUSD, targetHYPHA, targetHVOICE, targetSEEDS int64, env *Environment, t *testing.T) {
+
+	period, err := docgraph.GetLastDocumentOfEdge(env.ctx, &env.api, env.DAO, eos.Name("claimed"))
+	assert.NilError(t, err);
+
+	paymentEdges, err := docgraph.GetEdgesFromDocumentWithEdge(env.ctx, &env.api, env.DAO, period, eos.Name("payment"))
+	assert.NilError(t, err);
+
+	var paymentMap map[string]int64
+
+	paymentMap = make(map[string]int64)
+
+	for _, edge := range paymentEdges {
+		payment, err := docgraph.LoadDocument(env.ctx, &env.api, env.DAO, edge.ToNode.String())
+		assert.NilError(t, err);
+		asset, err := payment.GetContent("amount");
+		assert.NilError(t, err);
+		amount, symbol, err := AssetToInt(asset)
+		assert.NilError(t, err);
+		paymentMap[symbol] = amount
+	}
+
+	if (targetHUSD != 0) {
+		assert.Equal(t, targetHUSD, paymentMap["HUSD"])
+	}
+
+	if (targetHYPHA != 0) {
+		assert.Equal(t, targetHYPHA, paymentMap["HYPHA"])
+	}
+
+	if (targetHVOICE != 0) {
+		assert.Equal(t, targetHVOICE, paymentMap["HVOICE"])
+	}
+
+	if (targetSEEDS != 0) {
+		assert.Equal(t, targetSEEDS, paymentMap["SEEDS"])
+	}
+
+	assert.NilError(t, err);
+}
+
+func TestAdjustCommitment(t *testing.T) {
+	teardownTestCase := setupTestCase(t)
+	defer teardownTestCase(t)
+
+	env = SetupEnvironment(t)
+	t.Log(env.String())
+	t.Log("\nDAO Environment Setup complete\n")
+
+	// roles
+	proposer := env.Members[0]
+	assignee := env.Members[1]
+	closer := env.Members[2]
+
+	role1Doc := CreateRole(t, env, proposer, closer, role1)
+	t.Run("Test Adjust assignment commitment", func(t *testing.T) {
+
+		tests := []struct {
+			name       string
+			roleTitle  string
+			title      string
+			role       docgraph.Document
+			assignment string
+			husd       string
+			hypha      string
+			hvoice     string
+			usd        string
+		}{
+			{
+				name:       "role1 - 100% 100%",
+				roleTitle:  "Underwater Basketweaver",
+				title:      "Underwater Basketweaver - Atlantic",
+				role:       role1Doc,
+				assignment: assignment1,
+				husd:       "0.00 HUSD",
+				hypha:      "759.75 HYPHA",
+				hvoice:     "6078.02 HVOICE",
+				usd:        "3039.01 USD",
+			},
+		}
+
+		for _, test := range tests {
+
+			t.Log("\n\nStarting test: ", test.name)
+
+			_, err := dao.ProposeAssignment(env.ctx, &env.api, env.DAO, proposer.Member, assignee.Member, test.role.Hash, env.Periods[0].Hash, test.assignment)
+			assert.NilError(t, err)
+
+			// retrieve the document we just created
+			proposal, err := docgraph.GetLastDocumentOfEdge(env.ctx, &env.api, env.DAO, eos.Name("proposal"))
+			assert.NilError(t, err)
+		
+			ballot, err := proposal.GetContent("ballot_id")
+			assert.NilError(t, err)
+			voteToPassTD(t, env, ballot.Impl.(eos.Name))
```

Comment:
Heads up; This block is changing with #6 
`voteToPassTD` will receive slightly different params and the `ballot_id` won't exist.